pkg/events/derive/derive.go

```diff
@@ -5,26 +5,43 @@ import (
 	"github.com/aquasecurity/tracee/types/trace"
 )
 
-// deriveFunction is a function prototype for a function that receives an event as
+// DeriveFunction is a function prototype for a function that receives an event as
 // argument and may produce a new event if relevant.
 // It returns a derived or empty event, depending on successful derivation,
 // and an error if one occurred.
-type deriveFunction func(trace.Event) ([]trace.Event, []error)
+type DeriveFunction func(trace.Event) ([]trace.Event, []error)
 
 // Table defines a table between events and events they can be derived into corresponding to a deriveFunction
 // The Enabled flag is used in order to skip derivation of unneeded events.
 type Table map[events.ID]map[events.ID]struct {
-	DeriveFunction deriveFunction
-	Enabled        bool
+	DeriveFunction DeriveFunction
+	Enabled        func() bool
+}
+
+// Register registers a new derivation handler
+func (t Table) Register(deriveFrom, deriveTo events.ID, deriveCondition func() bool, deriveLogic DeriveFunction) {
+	if t[deriveFrom] == nil {
+		t[deriveFrom] = make(map[events.ID]struct {
+			DeriveFunction DeriveFunction
+			Enabled        func() bool
+		})
+	}
+	t[deriveFrom][deriveTo] = struct {
```

Comment:
Maybe we should handle the case that (deriveFrom, DeriveTo) pair is already registered and return an error in that case

Reply:
I thought I missed some validation here, thanks for catching it.